packages/rut/src/Element.ts

```diff
@@ -237,4 +242,22 @@ export default class Element<
   toString(): string {
     return this.name(true);
   }
+
+  async waitForPropChange<K extends keyof InferComponentProps<Type>>(prop: K) {
+    await act(async () => {
+      await new Promise((resolve, reject) => {
+        const interval = setInterval(() => {
+          if (this.element.props[prop as string] !== this[PROPS][prop as string]) {
+            clearInterval(interval);
+            resolve();
+          }
+        }, 50);
+
+        setTimeout(() => {
+          clearInterval(interval);
+          reject(new Error('Waiting for prop change has timed out.'));
+        });
```

Comment:
this will fire before the first interval, is this supposed to have a non-0 timeout?

Reply:
Yeah, oversight. Should be something like `10000`.